clicc_devices/views.py

```diff
@@ -80,3 +80,37 @@ def crontab(request: HttpRequest) -> HttpResponse:
         record, _ = CronJob.objects.get_or_create(pk=1)
         form = CronForm(instance=record)
     return render(request, "cron.html", {"form": form})
+
+
+def devices(request: HttpRequest) -> JsonResponse:
+    """Endpoint to retrieve all device data as JSON.
+    Data is grouped by Unit. Each Unit contains a dictionary of Item Types
+    and their counts.
+
+    Example response structure:
+    {
+        "unit1": {
+            "typeA": 17,
+            "typeB": 42,
+            },
+        "unit2": {
+            "typeA": 3,
+            "typeC": 12,
+            },
+    }
+
+    :param request: The HTTP request object.
+    :return: JSON response containing all device data.
+    """
+
+    all_sets = Set.objects.all().order_by("name")
```

Comment:
I think it would be better to sort by unit and item type name, not by set name.  The current set naming convention does give us the correct order for free, now, but that could change - it's out of our control within this application - so let's sort the data as well as we can based on how it will be used.